packages/plugin-core/src/components/lookup/vaultSelectionModeConfig.ts

```diff
@@ -0,0 +1,44 @@
+import { VaultSelectionMode } from "./types";
+import { ConfigUtils } from "@dendronhq/common-all";
+import { ExtensionProvider } from "../../ExtensionProvider";
+
+/**
+ * Class responsible for proxying interaction with vault
+ * selection mode configuration.
+ * */
+export class VaultSelectionModeConfig {
```

Comment:
write now, we have an existing convention for proxying with configuration. can we stick to that? see [ConfigUtils](https://github.com/dendronhq/dendron/blob/ec063f6a0c2e026056d34128f9eab3b0238b5f09/packages/common-all/src/utils.ts#L539)

if we want to make something that is workspace specific, create a new class there

Reply:
Hmm. My intuition is that `ConfigUtils` is more of an abstraction that pertains to retrieval of config values and what we have here is more closely coupled to _how vault selection works_. 

Of course, it's up to debate where we draw the distinction for each utility class' responsibility / abstraction level. My initial impression is that this class is closer to the vault selection implementation detail than a mechinism for config value retrieval.

Reply:
multiple things going on. when adding utility methods, two things to consider:
1. what package does it go under?
2. where in the package does it go under?

for 1, we always want a general utility to go in the most accessible location. so in order of preference:
- common-all
- common-server
- engine-server
- end client (eg. dendron-cli, plugin-core, etc)

so to answer 1, is this a utility that can be used across multiple clients and is not client specific? if so, it should be put into a `common-*` package

for 2, this depends on context. number one convention - when in doubt, go by what's already done. right now, configuration utilities (anything that is stricly getting a config or a derived config value) is handled in `ConfigUtils`.
furthermore, we already have a [PublishUtils](https://github.com/dendronhq/dendron/blob/763c797c4c2f7821ef747376c980e4a4b0eace8e/packages/common-all/src/utils.ts#L486) in the same file that calculates derived config values for publishing. so we should go with this convention here and add `VaultUtils` here.
some optimizations we can make:
- split off config utilities in a separate file
- make name more specific (eg. `PublishConfigUtils`)

Reply:
From levels of abstraction it would seem that `ConfigUtils` is a lower level abastraction than `VaultSelectionModeConfig` but if you guys want me to move the logic into `ConfigUtils` let me know.

Reply:
Renamed to VaultSelectionModeConfigUtils